### PR TITLE
Implement Led

### PR DESCRIPTION
**Pull request by daniel9a**
Defined and implemented led class.

### CODE REVIEW

File: Led.h
Comment (blhough) [inline Led.h]: By convention constants are all caps

File: Led.cpp
Comment (blhough) [inline Led.cpp]: Another requirement for the blink function was that if the led was already on from the user calling `turnOn` then an inverse blink would occur. Meaning it would turnoff for a short delay and then back on. Refer to the issue #46

File: Led.cpp
Comment (blhough) [inline Led.cpp]: To control brightness you will need to use analogWrite.

File: Led.cpp
Comment (blhough) [inline Led.cpp]: It would be best to convert `percent` here to the correct value range for when used by analogWrite

**Review by blhough — CHANGES_REQUESTED** [review]

File: Led.cpp
Comment (blhough) [inline Led.cpp]: This will never be true because your for-loop break condition requires `i<n`

**Review by blhough — CHANGES_REQUESTED** [review]

File: Led.h
Comment (blhough) [inline Led.h]: by convention constants are ALLCAPS

File: Led.h
Comment (blhough) [inline Led.h]: since `flag` can only have two values it would be best to use a bool.

File: Led.cpp
Comment (blhough) [inline Led.cpp]: This is correct but can you think of a way without relying on the built in function? Hint, since we are going from a range [0,A] to a range [0,B] this is a special case that resolves to just a constant multiplier.

File: Led.cpp
Comment (blhough) [inline Led.cpp]: the `Pin` inside the parenthesis are undefined because no parameter is passed in. Let default pin to 0 if no value is given.

File: Led.cpp
Comment (blhough) [inline Led.cpp]: if n == 1 then this loop executes twice which is not what we want. 
and n == 2 will blink 3 times and so on.

**Review by blhough — CHANGES_REQUESTED** [review]

File: Led.cpp
Comment (blhough) [inline Led.cpp]: `if(flag==true)` is the same as `if(flag)`

File: Led.h
Comment (blhough) [inline Led.h]: Let's be more descriptive than `flag`.
Use `Ctrl R, R` to rename the variable.

File: Led.cpp
Comment (blhough) [inline Led.cpp]: You can combine this and line 13.

File: Led.cpp
Comment (blhough) [inline Led.cpp]: If the led is already on then we need to change the active brightness
We do not want to turn it on every time using analogWrite.

File: Led.cpp
Comment (blhough) [inline Led.cpp]: since `i<=n` is never used just omit it.
`for( unsigned i = i;; i++)`
It will be clear to people reading the code the the break condition must be within the loop then.

**Review by blhough — CHANGES_REQUESTED** [review]

File: Led.cpp
Comment (blhough) [inline Led.cpp]: sort of, but there is no reason to assign the value to `percent`. Just multiply it and assign to `brightness`

File: Led.cpp
Comment (blhough) [inline Led.cpp]: No need for the `else`. Always look for ways to prevent repeating code

File: Led.cpp
Comment (blhough) [inline Led.cpp]: only if the led is on do we want to change the active brightness

**Review by blhough — CHANGES_REQUESTED** [review]

File: Led.cpp
Comment (blhough) [inline Led.cpp]: when I said "active brightness" I mean the brightness of the currently on led. If the led is on we need to change the active brightness. It it is on or off we still need to remember the brightness setting.

**Review by blhough — CHANGES_REQUESTED** [review]

File: Led.cpp
Comment (blhough) [inline Led.cpp]: regardless of if the led is on or not we need to save the new brightness value

**Review by blhough — CHANGES_REQUESTED** [review]

**Comment by blhough** [comment]
Looks good. Let's to wait until the ArduinoSpoof #51 is merged into `dev`. Then update from `dev` and un-comment and double check the arduino functions.

**Comment by daniel9a** [comment]
Sounds good to me. If I can do anything else I can do to stay productive in the time being please let me know captain.

**Comment by blhough** [comment]
Okay go ahead and update from `dev` now

**Comment by blhough** [comment]
You have a good start on the ledRGB. Let's finish this one up first so the rest of the rgb will be more clear.
first make sure sure all your changes are committed and then checkout this branch.
then `git pull origin master` to update this branch from dev.
The update will add a new file called `Arduino.h`.
`#include "Arduino.h"` in the cpp file to allow use of the arduino specific functions.
Uncomment and double checkout your uses of the functions, remembering to test compile.
Then commit and push to this branch again to update.

**Comment by daniel9a** [comment]
I tried doing this but when I pushed I keep getting this:


$ git push origin led
To https://github.com/RoboticsClubAtUCDavis/micromouse.git
 ! [rejected]        led -> led (non-fast-forward)
error: failed to push some refs to 'https://github.com/RoboticsClubAtUCDavis/mic                                                                                                                                  romouse.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

**Comment by blhough** [comment]
hmm not sure how that happened. Go ahead and make a new branch off of `led`
`git checkout -b led2` and then push that to github

**Review by blhough — APPROVED** [review]

**Pull request merged**
